dotCMS/src/main/java/com/dotcms/rendering/velocity/services/ContentletLoader.java

```diff
@@ -581,8 +581,8 @@ public InputStream writeObject(final VelocityResourceKey key)
 
     }
 
-    private boolean isLiveVersionAvailable(VelocityResourceKey key, ContentletVersionInfo info) {
-        return info != null && key.mode.showLive && UtilMethods.isSet(info.getLiveInode());
+    private boolean isLiveVersionNotAvailable(VelocityResourceKey key, ContentletVersionInfo info) {
+        return info == null || key.mode.showLive && !UtilMethods.isSet(info.getLiveInode());
```

Comment:
I am wondering if this should be something like
```
        return info == null || !key.mode.showLive || !UtilMethods.isSet(info.getLiveInode());
```
Also set final the parameters

Reply:
I am not sure because not make sense to me this  one "!key.mode.showLive" nor  "!UtilMethods.isSet(info.getLiveInode())" neither.

I think it should be something like  (according to the name):

```suggestion
        DotPreconditions.checkNotNull(info);
        return key.mode.showLive && UtilMethods.isSet(info.getLiveInode());
```